File: C++/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) // if strs is empty
            return "";
        string prefix = strs[0];
        for (int i = 1; i < strs.size(); i++) {
            if (strs[i].size() < prefix.size()) // make prefix be the shortest one
                prefix = strs[i];
        }

        for (int i = 0; i < strs.size(); i++) {
            for (int j = 0; j < prefix.size(); j++) {
                if (prefix[j] != strs[i][j])
                    prefix =  prefix.substr(0, j); // cut off the  uncommon prefix
            }
        }
        return prefix;
    }
};

File: 算法/Python/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:  # if strs is empty
            return ""

        prefix = strs[0]  # make prefix be the shortest one
        for str_ in strs:
            if len(str_) < len(prefix):
                prefix = str_

        for str_ in strs:  # find the common prefix
            for i in range(len(prefix)):
                if prefix[i] != str_[i]:
                    prefix = prefix[:i]
                    break
        return prefix
